packages/cosmos/src/tokenfactory.rs

```diff
@@ -0,0 +1,243 @@
+use crate::{Cosmos, TypedMessage, Wallet};
+use anyhow::{Context, Result};
+use cosmos_sdk_proto::cosmos::{
+    bank::v1beta1::Metadata,
+    base::{abci::v1beta1::TxResponse, v1beta1::Coin},
+};
+use prost::Message;
+
+/// TokenFactory interface
+pub struct TokenFactory<'a, 'b> {
+    client: &'a Cosmos,
```

Comment:
I think it's fine to simply clone the `Cosmos` to avoid the reference complexity, it uses an `Arc` internally. And it may be worth making a similar change to `Wallet` so that it can be cheaply cloned as well. Neither of these are requirements for this PR though.

Reply:
ah, nice, I wasn't familiar with deadpool (the crate) :) changed in https://github.com/Levana-Protocol/levana-cosmos-rs/pull/5/commits/84d28a9ec79453f71ca6a8515180eec725bcf110

Reply:
Oh, hah, I totally forgot I'd already moved us over to deadpool.